lib/request_overrider.js

```diff
@@ -227,6 +227,13 @@ function RequestOverrider(req, options, interceptors, remove, cb) {
     /// like to change request.path in mid-flight.
     options.path = req.path
 
+    // similarly, node-http-proxy will modify headers in flight, so we have to put the headers back
+    // into options
+    const reqHeaders = req.getHeaders ? req.getHeaders() : req._headers
+    if (Object.keys(reqHeaders).length > 0) {
+      options.headers = req.getHeaders ? req.getHeaders() : req._headers
+    }
```

Comment:
Can this be replaced with `options.headers = req.getHeaders()`? The `req.getHeaders` truthiness check is unnecessary in the Node versions we support. That would also fix the coverage issue you mentioned.

Reply:
Yep sounds simpler - the only problem is that is causes some of the other tests that are expecting `headers: undefined` to fail because they now get `headers: {}`, but that's easy enough to change in the test if you're happy with it :).